src/toil/fileStore.py

```diff
@@ -721,6 +721,13 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                                                               0.0, False)
         return localFilePath
 
+    def exportFile(self, jobStoreFileID, dstUrl):
+        while jobStoreFileID in self._pendingFileWrites:
```

Comment:
Cleaner, maybe. But i feel that since the wait is inherently dependent on caching, i feel that would be the correct way to do it.